modules/packages/Sort.chpl

```diff
@@ -606,15 +606,15 @@ private proc _MergeSort(Data: [?Dom], minlen=16, comparator:?rec=defaultComparat
     return;
   }
   const mid = (hi-lo)/2+lo;
-  var A1 = Data[lo..mid];
-  var A2 = Data[mid+1..hi];
+  ref A1 = Data[lo..mid];
+  ref A2 = Data[mid+1..hi];
   cobegin {
     { _MergeSort(A1, minlen, comparator); }
     { _MergeSort(A2, minlen, comparator); }
   }
 
   // TODO -- This iterator causes unnecessary overhead - we can do without it
-  for (a, _a) in zip(Data[lo..hi], _MergeIterator(A1, A2, comparator=comparator)) do a = _a;
+  for (a, _a) in zip(Data, _MergeIterator(A1, A2, comparator=comparator)) do a = _a;
```

Comment:
Looking at the _MergeIterator iter -- it can't be zippered inlined, so it might be interesting to see what sort of performance we could get by manually inlining it. Or maybe making it non-zippered by doing something like: 

```chpl 
  var i = lo;
  for _a in _MergeIterator(A1, A2, comparator=comparator) {
    Data[i] = _a;
    i += 1;
  }
```

Probably more than you want to look into for now, but I couldn't understand why your change would hurt performance so I was digging around a little.